sktime/performance_metrics/forecasting/probabilistic/_classes.py

```diff
@@ -55,25 +59,31 @@ def __call__(self, y_true, y_pred, **kwargs):
         """
         return self.evaluate(y_true, y_pred, multioutput=self.multioutput, **kwargs)
 
-    def evaluate(self, y_true, y_pred, multioutput, **kwargs):
+    def evaluate(self, y_true, y_pred, multioutput=None, **kwargs):
         """Evaluate the desired metric on given inputs."""
-        # Input checks
-        _, y_true, y_pred, multioutput = _check_reg_targets(y_true, y_pred, multioutput)
-        # y_pred = _check_y_pred_type()
-
-        return self._evaluate(y_true, y_pred, multioutput, **kwargs)
+        # Input checks and conversions
+        y_true_inner, y_pred_inner, multioutput = self._check_ys(
+            y_true, y_pred, multioutput
+        )
+        # pass to inner function
+        return self._evaluate(y_true_inner, y_pred_inner, multioutput, **kwargs)
 
     def _evaluate(self, y_true, y_pred, multioutput, **kwargs):
+        # todo: should have a default implementation, which is just the mean over
+        #       _evaluate_by_index
```

Comment:
I have done this, however as the default implementations both end up calling each other, at least one of them needs to be implemented manually to avoid a recursion error (I've added a check for this case.)

Reply:
correct - something similar happens for `predict_interval` etc in the `BaseForecaster`, and in the design here:
https://github.com/alan-turing-institute/sktime/pull/2100
if one is implemented, the others do sth sensible; if none is, an error message is raised. You can use `_has_implementation` if you are inheriting from `BaseObject`